src/BizHawk.Emulation.Cores/Consoles/Nintendo/Gameboy/Gambatte.cs

```diff
@@ -22,8 +22,7 @@ public partial class Gameboy : IInputPollable, IRomInfo, IGameboyCommon, ICycleT
 
 		[CoreConstructor(VSystemID.Raw.GB)]
 		[CoreConstructor(VSystemID.Raw.GBC)]
-		[CoreConstructor(VSystemID.Raw.SGB)]

```

Comment:
Why is SGB being removed here? BSNES/BSNESv115 still have it.

Reply:
Because for core choosing, the system is only set to `SGB` when the core is bsnes: https://github.com/TASEmulators/BizHawk/blob/9ee38301aff5b0322c9e10dfc3ecce75a5304c1e/src/BizHawk.Client.Common/RomLoader.cs#L485-L490

Reply:
How does this exactly end up working for say, loading a GBC only ROM into BSNES. Does it proceed to fail completely due to not finding a fallback with the SGB system ID ctor, or does it end up still picking Gambatte with the GBC system ID?

On that, the code tries to handle old BSNES, yet it's not added into the config selection (for whatever reason).

Reply:
That's a good point. It does NOT work currently for that exact reason and instead runs into this code whichthen no longer makes sense: https://github.com/TASEmulators/BizHawk/blob/9ee38301aff5b0322c9e10dfc3ecce75a5304c1e/src/BizHawk.Client.Common/RomLoader.cs#L859

Reply:
Unrelated to the above bug, I think these `[CoreConstructor]`s should be kept, so that `emu.SystemId` always matches a `[CoreConstructor]`.

Reply:
There's more to this, and there is already broken behavior currently. I'll make an issue later.